snforge_std/src/cheatcodes/tx_info.cairo

```diff
@@ -1,83 +1,49 @@
-#[derive(Copy, Drop, Serde)]
-struct TxInfoMock {
-    version: Option<felt252>,
-    account_contract_address: Option<ContractAddress>,
-    max_fee: Option<u128>,
-    signature: Option<Span<felt252>>,
-    transaction_hash: Option<felt252>,
-    chain_id: Option<felt252>,
-    nonce: Option<felt252>,
-    // starknet::info::v2::TxInfo fields
-    resource_bounds: Option<Span<ResourceBounds>>,
-    tip: Option<u128>,
-    paymaster_data: Option<Span<felt252>>,
-    nonce_data_availability_mode: Option<u32>,
-    fee_data_availability_mode: Option<u32>,
-    account_deployment_data: Option<Span<felt252>>,
-}
-
-trait TxInfoMockTrait {
-    /// Returns a default object initialized with Option::None for each field
-    /// Useful for setting only a few of fields instead of all of them
-    fn default() -> TxInfoMock;
-}
-
-impl TxInfoMockImpl of TxInfoMockTrait {
-    fn default() -> TxInfoMock {
-        TxInfoMock {
-            version: Option::None(()),
-            account_contract_address: Option::None(()),
-            max_fee: Option::None(()),
-            signature: Option::None(()),
-            transaction_hash: Option::None(()),
-            chain_id: Option::None(()),
-            nonce: Option::None(()),
-            resource_bounds: Option::None(()),
-            tip: Option::None(()),
-            paymaster_data: Option::None(()),
-            nonce_data_availability_mode: Option::None(()),
-            fee_data_availability_mode: Option::None(()),
-            account_deployment_data: Option::None(()),
-        }
-    }
-}
-
 /// Changes `TxInfo` returned by `get_tx_info()` for the targeted contract and span.
-/// - `target` - instance of `CheatTarget` specifying which contracts to spoof
 /// - `tx_info_mock` - a struct with same structure as `TxInfo` (returned by `get_tx_info()`)
-/// - `span` - instance of `CheatSpan` specifying the number of target calls with the cheat applied
-fn spoof(target: CheatTarget, tx_info_mock: TxInfoMock, span: CheatSpan) {
-    validate_cheat_target_and_span(@target, @span);
+fn spoof(tx_info_mock: TxInfoMock) {
+    let mut execution_info: ExecutionInfoMock = Default::default();
+
+    execution_info.tx_info = tx_info_mock;
 
-    let mut inputs = array![];
-    target.serialize(ref inputs);
-    span.serialize(ref inputs);
-    tx_info_mock.serialize(ref inputs);
-    handle_cheatcode(cheatcode::<'spoof'>(inputs.span()));
+    cheat_execution_info(execution_info);
 }
 
 /// Changes `TxInfo` returned by `get_tx_info()` for the targeted contract until the spoof is canceled with `stop_spoof`.
-/// - `target` - instance of `CheatTarget` specifying which contracts to spoof
 /// - `tx_info_mock` - a struct with same structure as `TxInfo` (returned by `get_tx_info()`)
-fn start_spoof(target: CheatTarget, tx_info_mock: TxInfoMock) {
-    spoof(target, tx_info_mock, CheatSpan::Indefinite);
+fn start_spoof(tx_info_mock: TxInfoMock) {
+    spoof(tx_info_mock);
```

Comment:
so we have 2 functions that do the same?

Reply:
Probably, it's fixed later